articles/remapping.md

```diff
@@ -0,0 +1,77 @@
+
+#### Requirements for remapping capabilities
+##### 1. Namespace Replacement
+**A user can remap the first namespace token without replacing all namespace tokens with the same name.**
+
+*Example*
+Node uses names `"/big/cat"`, `"/big/red/dog"`, `"/red/big/cat"`. It must be possible to make a rule to that replaces "big" with "small" such that the final names are `"/small/cat"`, `"/small/red/dog"`, `"/red/big/cat"`
+
+**A user can replace all matching namespace tokens without changing the basename.**
+
+*Example*
+Node uses names `"/big/cat"`, `"/red/big/dog"`, `"/dog/big"`. It must be possible to make a rule to that replaces "big" with "small" such that the final names are `"/small/cat"`, `"/red/small/dog"`, `"/dog/big"`
+
+**A user can replace multiple matching namespace tokens only if whole namespaces are matched.**
+
+*Example*
+Node uses names `"/big/red/dog"`, `"/very_big/red/dog"`. It must be possible to make a rule to that replaces "big/red" with "small" such that the final names are `"/small/dog"`, `"/very_big/red/dog"`
+
+##### 2. Basename Replacement
+**A user can replace a full basename token with another basename token**
+
+*Example*
+Node uses names `"/big/cat"`, `"/big/cat/paws"`. It must be possible to make a rule to that replaces "cat" with "dog" such that the final names are `"/big/dog"`, `"/big/cat/paws"`
+
+##### 3. Exact Replacement
+**A user can replace a full name with another name**
+
+*Example*
+Node uses names `"/big/cat"`, `"/big/cat/paws"`. It must be possible to make a rule to that replaces "/big/cat" with "/bear" such that the final names are `"/bear"`, `"/big/cat/paws"`
+
+##### 4. Arbitrary Substring Replacement
+**A user can replace any substring with another string**
+
+*Examples*
+Node uses names `"/big/red/dog"`, `"/very_big/red/dog"`. It must be possible to make a rule to that replaces "big/red" with "small" such that the final names are `"/small/dog"`, `"/very_small/dog"`
+
+Node uses names `"/big/dog"`, `"/dog/big"`. It must be possible to make a rule to that replaces "dog" with "cat" such that the final names are `"/big/cat"`, `"/cat/big"`
+
+Node uses names `"/big/red/dog"`. It must be possible to make a rule to that replaces "/" with "_" such that the final name is `"/_big_red_dog"`
+
+#### Static and Dynamic Remapping
+Static remapping is the ability to provide a node with remap rules prior to launching it. A node stores this remapping rules for the duration of it's life. Any names it uses get have these rules applied to it before it is used. 
+
+Dynamic remapping is the ability to remap a name after it has already been used. For a neat user experience, Dynamic remapping requires rules to be changed. A user should be able to remap the name they see rather than the original name that was used. This suggests remapping rules have an order in which they should be applied (chronological?).
```

Comment:
Maybe? This was something we were debating amongst ourselves during the meeting. I personally believe you need to at least store the "static" remappings for when you create new entities that use "Names" (as you mention above), but it's less clear to me about the dynamic remappings. We need to first decide what kinds of remappings may be used in dynamic remapping and if they apply to all existing things, all new things, or are scoped more narrowly than static remappings (i.e. applies to a single entity versus a whole node).